src/backends/meta-monitor-config-manager.c

```diff
@@ -234,6 +234,39 @@ assign_monitor_crtcs (MetaMonitorManager       *manager,
     }
 
   monitor_mode = meta_monitor_get_mode_from_spec (monitor, monitor_mode_spec);
+  if (!monitor_mode)
+    {
+      MetaOutput *current_output = meta_monitor_get_main_output(monitor);
+
+      /* When switching from underscanning to non-underscanning modes, the
+       * target modes from the configuration won't match any of the currently
+       * available modes and no valid mode will be found, so we need to adjust
+       * the values of the mode we are searching for to account for that.
+       *
+       * Since we can't check from here whether we're using the normal implementation
+       * of underscanning (i.e. using 'on' for the 'underscan' atom, instead of 'crop'),
+       * we do these additional checks once we have failed to obtain a valid mode, so
+       * that this works when using either the armsoc and the xf86-video-intel drivers.
+       */
+      if (current_output->is_underscanning != monitor_config->enable_underscanning)
+        {
+          MetaMonitorModeSpec *underscanning_mode_spec = g_memdup (monitor_mode_spec, sizeof (MetaMonitorModeSpec));
```

Comment:
Is it possible to use `g_autofree` here?

Reply:
Since `g_memdup` returns memory allocated with `g_malloc`, that should be possible. Yes.